rust/lance-datafusion/Cargo.toml

```diff
@@ -32,7 +32,8 @@ snafu.workspace = true
 tokio.workspace = true
 
 [dev-dependencies]
-substrait-expr = { version = "0.2.1" }
+# TODO: This is too old
```

Comment:
I believe somebody needs to also update substrait-expr to use prost 0.13 to match the version used by datafusion-substrait

Reply:
Guess that's on me.  How do you feel about bumping all the way to datafusion 43?  That release should include a change I made which will let us drop substrait-expr entirely.

Reply:
Version 0.2.2 of substrait-expr is now available and will include prost 0.13

Reply:
>  How do you feel about bumping all the way to datafusion 43

There were a couple more API surfaces that needed modifying so I was focusing on min-change possible, but I don't have a philosophiscal objection.

Reply:
Ok.  Let's keep it minimal for now.  I'll need to do some more work to get rid of substrait-expr anyways.